version.go

```diff
@@ -25,7 +26,9 @@ type versionsByNumber []*Version
 func (s versionsByNumber) Len() int      { return len(s) }
 func (s versionsByNumber) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s versionsByNumber) Less(i, j int) bool {
-	return s[i].Number < s[j].Number
```

Comment:
If we choose to be Go 1.8+, we can shorten this a bit to:

```go
sort.Slice(versions, func(i, j int) bool {
  return versions[i].Number < versions[j].Number
})
```

https://golang.org/pkg/sort/#Slice

What's weird is that Fastly's API clearly says that  "Number" is an integer, yet I tested it with curl and it's most definitely not. I wonder if there's a larger issue here that's fixable using mapstructure's weakdecode. We might be able to decode the string into an int automatically, and then the sort becomes a very simple less function.